### ISSUE TITLE
DTDT_moist does not seem that DTDT (possible issue with NCEP GWD)

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/GEOSgcm_GridComp#252]
Per an email from Julio Bacmeister, the variable `DTDT_moist` doesn't seem that DTDT:

> I've been trying to hook up the NCAR convective gravity wave scheme in GEOS.  For the heating profile I was going to use the DTDT_moist variable that is used by the NCEP GW routines.  However, when I look at profiles of DTDT_moist in the GW scheme it looks disturbingly like temperature.  And when I look in GEOS_MoistGridComp.F90 I see a line:
> 
>  if(associated(DTDT_MOIST)) DTDT_MOIST   = TEMP
> 
> but NO corresponding 
> 
>  DTDT_moist=  (TEMP-DTDT_moist)/DT_MOIST
> 
> like there is for DTDT_macro for example.
> 
> I'd add the tendency line myself, but GEOS_MoistGridComp.F90 has gotten quite big and complex, and I'm not sure where it should go.  

To wit, let's look at `DTDT_macro`:
```
6930:         if(ALLOC_DTDT_macro ) allocate(DTDT_macro(IM,JM,LM))
8502:        DTDT_macro=TEMP
8541:        DTDT_macro=  (TEMP-DTDT_macro)/DT_MOIST
8893:         DTDT_macro=TEMP
8997:         DTDT_macro=  (TEMP-DTDT_macro)/DT_MOIST
10131:         DTDT_macro =  TEMP
10450:         DTDT_macro=TEMP
10595:         DTDT_macro=  (TEMP-DTDT_macro)/DT_MOIST
12500:         if(ALLOC_DTDT_macro   )  deallocate(DTDT_macro  )
```
Usually, a `DTDT_macro` setting to `TEMP`, is delta'd and then divided by DT. (The one at 10131 seems a bit unbalanced, but it's hard to follow Moist GC with all it's ifs. I assume `DTDT_micro` is okay?)

Now let's look at `DTDT_moist`:
```
7045:         if(ALLOC_DTDT_moist    ) allocate(DTDT_moist    (IM,JM,LM))
7341:      if(associated(DTDT_MOIST)) DTDT_MOIST   = TEMP
12543:         if(ALLOC_DTDT_moist  )   deallocate(DTDT_moist )
```
Well, that's not right. 

And, as Julio says, `DTDT_moist` is used as an import to GWD (where it fills `TRATE`).

I'm assigning it to, well, anyone I can think of that knows Moist. Hopefully one of you can take a look.

**Comment by mathomp4** [comment]
Also, pinging @sdrabenh as a notification as the Gatekeeper.

**Comment by narnold1** [comment]
Confirmed that DTDT_MOIST is missing a follow-up, and is equal to temperature.  I'll push a fix this afternoon.  
Note there is (what should be) an equivalent variable, "TIM", from PHYSICS.

**Comment by mathomp4** [comment]
@narnold1 Do you think there is a missing follow-up for `DTDT_macro` as well? I'm only concerned about how there are two set-to-`TEMP` in a row in the GC:
```
10131:         DTDT_macro =  TEMP
10450:         DTDT_macro=TEMP
10595:         DTDT_macro=  (TEMP-DTDT_macro)/DT_MOIST
```
Or can you follow the if's and know it's okay?

**Comment by dbarahon** [comment]
DTDT_macro comes in pairs. These are all right
8502 and 8541
8893 and 8997
10450 and 10595
****10131 is redundant

Donifan
